### DISCUSSION TITLE
Show backtraces in WebUI

### DISCUSSION THREAD

**Discussion by amo13** [bensheldon/good_job#1123] [Q&A] [answered]
When a job fails, it shows as discarded in the webUI and displays the error with the error message. Is it possible to display the backtrace in addition to the error message?

![image](https://github.com/bensheldon/good_job/assets/6617661/c8099d77-d684-4ad4-893f-5d3130ffc6a4)

**Comment by bensheldon** [comment, marked as answer]
I'm sorta ambivalent to storing/displaying/feature-ing backtraces. If someone made the PR, I imagine I'd accept it, but it's low on my own priority list to build into GoodJob.

My reasoning is:

- Capturing error classes is part of Active Job, and core to how Active Job handles and configures retries. That's why they exist in GoodJob.
- Backtraces feel more like generalized debugging, and I believe that anyone who is debugging backtraces for jobs also needs to debug backtraces for web requests and rake tasks, etc. So seems like using Sentry/Bugsnag/etc. _generally_ would be better.
- When I do think about what it would require to support backtraces, it gets.... complicated.   Which is why I sort of also would rather someone else do it. Particularly: how to store them? (blob of text? more structured? would it need to be duplicated across both jobs and individual executions) would it need to be parsed say, to separate filepaths and line numbers? should it be passed through the Rails backtrace filter? how to handle exception `cause` / nested exceptions? Seems like a lot of design to figure out.
- It overlaps a bit with logs/logging, which also has been low on my priority list working on (#733) 

That all said, I'm open if someone else were to work on it. I'd also be curious if that would be something that should live outside of GoodJob; seems like it would be nice to have a Rails Engine that could be an exception and log and metrics sink for its parent application.

**Reply by amo13** [reply]
Thank you for your thoughts on this. I find that all pretty convincing. And I wouldn't want to encourage the urgent development of such a feature after reading your reasoning.
